libflux/flog: remove unnecessary + 1 on buffers

Noticed in libflux/flog that code like this is used a lot of places.

```
    char buf[FLUX_MAX_LOGBUF + 1];
    (void)vsnprintf (buf, sizeof (buf), fmt, ap);
```

Typically the +1 is used to ensure that the buffer will always be NUL terminated, since `snprintf()` (and family) may not NUL terminate.  But if the size of the buffer passed to `snprintf()` is always `sizeof (buf)`, then it doesn't matter.  Instead `FLUX_MAX_LOGBUF` should be passed in.

Edit:

Not sure what would be best.  Probably?

```
char buf[LOGBUF + 1];
(void)vsnprintf (buf, LOGBUF, fmt, ap);
buf[LOGBUF] = '\0';
```

Here's my experiment :-)
```c
#include <assert.h>
#include <stdio.h>

int main(int argc, char **argv)
{
        char buf[4] = { 'x', 'x', 'x', 'x' };
        int len = snprintf (buf, sizeof (buf), "%s", "0123456789");

        printf ("snprintf returned %d\n", len);

        assert (buf[0] == '0');
        assert (buf[1] == '1');
        assert (buf[2] == '2');
        assert (buf[3] == '\0');
}
```

```
$ cc     foo.c   -o foo
foo.c: In function ‘main’:
foo.c:7:42: warning: ‘%s’ directive output truncated writing 10 bytes into a region of size 4 [-Wformat-truncation=]
    7 |  int len = snprintf (buf, sizeof (buf), "%s", "0123456789");
      |                                          ^~   ~~~~~~~~~~~~
foo.c:7:12: note: ‘snprintf’ output 11 bytes into a destination of size 4
    7 |  int len = snprintf (buf, sizeof (buf), "%s", "0123456789");
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
$ ./foo
snprintf returned 10
$
```

--- a/src/common/libflux/flog.c
+++ b/src/common/libflux/flog.c
@@ -33,7 +33,7 @@
 typedef struct {
     char appname[STDLOG_MAX_APPNAME + 1];
     char procid[STDLOG_MAX_PROCID + 1];
-    char buf[FLUX_MAX_LOGBUF + 1];
+    char buf[FLUX_MAX_LOGBUF];
     flux_log_f cb;
     void *cb_arg;
 } logctx_t;
@@ -128,7 +128,7 @@ int flux_vlog (flux_t *h, int level, const char *fmt, va_list ap)
     char *xtra = NULL;
 
     if (!h) {
-        char buf[FLUX_MAX_LOGBUF + 1];
+        char buf[FLUX_MAX_LOGBUF];
         const char *lstr = stdlog_severity_to_string (LOG_PRI (level));
 
         (void)vsnprintf (buf, sizeof (buf), fmt, ap);
@@ -193,7 +193,7 @@ int flux_log (flux_t *h, int lev, const char *fmt, ...)
 void flux_log_verror (flux_t *h, const char *fmt, va_list ap)
 {
     int saved_errno = errno;
-    char buf[FLUX_MAX_LOGBUF + 1];
+    char buf[FLUX_MAX_LOGBUF];
 
     (void)vsnprintf (buf, sizeof (buf), fmt, ap);
     flux_log (h, LOG_ERR, "%s: %s", buf, flux_strerror (saved_errno));

